AirCasting/Settings/SettingsMyAccountViewModel.swift

```diff
@@ -0,0 +1,80 @@
+// Created by Lunar on 16/06/2022.
+//
+
+import Foundation
+import Resolver
+
+final class SettingsMyAccountViewModel: ObservableObject {
+    @Injected private var networkChecker: NetworkChecker
+    @Injected private var sessionEntityStore: SessionEntityStore
+    @Injected private var logoutController: LogoutController
+    @Injected private var deleteController: DeleteAccountController
+    @InjectedObject private var userState: UserState
+    @Published var alert: AlertInfo?
+    
+    func signOutButtonTapped() {
+        guard networkChecker.connectionAvailable else {
+            showAlert(InAppAlerts.unableToLogOutAlert())
+            return
+        }
+        
+        do {
+            let result = try sessionEntityStore.anyLocationlessSessionsPresent()
+            switch result {
+            case true:
+                self.showAlert(InAppAlerts.logoutWarningAlert {
+                    self.logoutUser()
+                })
+            case false: self.logoutUser()
+            }
+        } catch {
+            Log.error("Error when informing the user about loosing locationless sessions")
+            self.showAlert(InAppAlerts.failedFetchingLocationlessSessionsAlert())
+        }
+    }
+    
+    func deleteButtonTapped() {
+        showAlert(InAppAlerts.firstConfirmationDeletingAccountAlert {
+            self.showAlert(InAppAlerts.secondConfirmationDeletingAccountAlert {
+                guard self.networkChecker.connectionAvailable else {
+                    self.showAlert(InAppAlerts.unableToConnectBeforeDeletingAccount())
+                    return
+                }
+                self.changeUserState(to: .deletingAccount)
+                self.deleteController.deleteAccount { result in
+                    self.changeUserState(to: .idle)
+                    switch result {
+                    case .success(_): break
+                    case .failure(let error):
+                        self.showAlert(InAppAlerts.failedDeletingAccount())
+                        assertionFailure("Failed to delete account: \(error)")
```

Comment:
we shouldn't do assertion here. Asserts are designed for defining expectations (asserting ;) ), usually guarding ourselves from programming errors, like here:
```Swift
func divide(_ number: Int, by divider: Int) -> Double {
    assert(divider != 0)
    return ...
}
```

this `deleteButtonTapped()` code in here is a runtime failure condition that is not caused by programming error, it can just happen due to outside conditions. In this case I think an error log would be sufficient